File: backend/app/schemas/resume.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict


class ResumeArtifactResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    user_id: str
    type: str
    latex_source: str
    pdf_storage_reference: Optional[str] = None
    page_count: int
    status: str
    error_message: Optional[str] = None
    created_at: datetime
    updated_at: datetime


class ResumeGenerateResponse(BaseModel):
    status: str
    message: str
    page_count: int
    pdf_storage_reference: Optional[str] = None
    artifact: Optional[ResumeArtifactResponse] = None

File: frontend/src/types/resume.ts
export type ResumeArtifactStatus = "success" | "overflow" | "compiler_unavailable" | "error" | "draft";

export interface ResumeArtifact {
  id: string;
  user_id: string;
  type: "base" | string;
  latex_source: string;
  pdf_storage_reference?: string | null;
  page_count: number;
  status: ResumeArtifactStatus;
  error_message?: string | null;
  created_at: string;
  updated_at: string;
}

export interface ResumeGenerateResponse {
  status: ResumeArtifactStatus;
  message: string;
  page_count: number;
  pdf_storage_reference?: string | null;
  artifact?: ResumeArtifact;
}
